### DISCUSSION TITLE
Boolean inputs act differently in workflow_dispatch and workflow_call

### DISCUSSION THREAD

**Discussion by AvivPeledTalon** [community/community#9343] [Actions] [answered]
Hello,

It seems like boolean inputs act as strings when used in a workflow_dispatch event, and as actual boolean variables when used in a reusable workflow (workflow_call).

For example, in one of my workflows, which acts on a workflow_dispatch event I use a boolean input "should_run_tests",  which controls the tests step using an if condition on the step.
In order for it to behave as intended, I have to use the following condition
`if: ${{ github.event.inputs.should_run_tests == "true" }}`

And not as you would expect from a boolean:
`if: ${{ github.event.inputs.should_run_tests }}`

It seems like the boolean type affects only the input form because I can see a checkbox when trying to run the workflow, but the value passed to the workflow is the string representation "true" or "false".

Doing the same thing in a workflow_call event, the boolean values that are being passed from one workflow to the other really seem to be booleans, and can be tested as such without string comparison, which I think is the right way to handle these and that should be the same in workflow_dispatch events

**Comment by NOUIY** [comment]
https://docs.microsoft.com/teamblog/introducing-docs-microsoft-com/

**Comment by ethomson** [comment]
Thanks @AvivPeledTalon - I agree that this difference is a bit confusing.  The reason is because when `workflow_dispatch` was conceived it was all strings, and it learned about boolean types after-the-fact.  To allow people to declare existing "boolean" inputs as actual boolean inputs, they remained strings.

The team is taking a look at some ways that we can improve this without breaking workflows that rely on it.  Hopefully we'll have more soon!

**Reply by fodbub-dukfU9-gimgug** [reply]
https://info.privacy.schibsted.com/se/login-se/

**Reply by mchascobarry** [reply]
Have this been fixed already, or it is still the same behaviour?

**Comment by ajaykn** [comment, marked as answer]
@AvivPeledTalon We have rolled out improvements and here is the blog post
https://github.blog/changelog/2022-06-10-github-actions-inputs-unified-across-manual-and-reusable-workflows/

Now we should be able to see dispatch inputs are available in both `github.event.inputs` and `inputs` context.
Where as now inputs context is recommended to use, because we respect boolean types.
cc @chrispat @avocaddo

**Reply by AvivPeledTalon** [reply]
That's even better! Thanks!

**Comment by victor-chan-groundswell** [comment]
hi @ajaykn ,

I'm actually trying to debug something similar where this is not working quite right.  

I have a workflow that is callable from either workflow_call or workflow_dispatch.  As mentioned I have a lot of fields where I do ${{ github.event.inputs.environment || inputs.environment }} to make sure that the inputs work from either the manual call or call from another workflow.

However, one step in my action, where it's I have this condition 
${{ github.event.inputs.environment || inputs.environment }} != 'dev' does not work when it's being called from a workflow_call, but it works when it's getting directly being called using workflow_dispatch.

**Comment by tomsit-ionos** [comment]
see also https://github.com/actions/runner/issues/3571
